src/main/java/com/scalar/cassy/transferer/AzureBlobFileUploader.java

```diff
@@ -40,57 +46,83 @@ public AzureBlobFileUploader(BlobContainerAsyncClient blobContainerClient) {
     }
 
     logger.info("Uploading " + file);
-    return blobContainerClient
-        .getBlobAsyncClient(key)
-        .uploadFromFile(file.toString(), true)
-        .toFuture();
+    return executorService.submit(
+        () -> {
+          try {
+            try (InputStream inputStream = new FileInputStream(file.toString())) {
+              blobContainerClient
+                  .getBlobClient(key)
+                  .upload(inputStream, new File(file.toString()).length(), true);
+            }
+          } catch (IOException e) {
+            throw new FileTransferException(e);
+          }
+
+          logger.info("Upload file succeeded : " + file.toString());
```

Comment:
You can omit `.toString()`.

Reply:
Thank you for your suggestion.
Sorry, Log removed because it is available in the upload method f8d33d6bc1766c1d8288ac71de90623d8b152aa3